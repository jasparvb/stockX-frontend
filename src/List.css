.List .list-box {
  border: 1px solid #ced4da;
  border-radius: 3px;
  padding: 10px 10px 20px;
}