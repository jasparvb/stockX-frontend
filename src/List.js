import React from "react";
import ListItem from './ListItem';
import './List.css';

function List({id, name, stocks, removeList, removeStock}) {

  return (
    <div className="List col-md-4 col-lg-4 mb-4">
      <div className="list-box">
        <h3 className="mb-4">{name}</h3>
        {stocks.length ? 
          stocks.map(s => (
            <ListItem 
              key={s.id} 
              id={s.id} 
              ticker={s.ticker} 
              name={s.name} 
              quote={s.quote}
              listId={id} 
              removeStock={removeStock} 
            />
          ))
        : <p>No stocks added yet</p>
        }
        <button className="btn btn-primary btn-sm mt-3" onClick={() => removeList(id)}>Delete List</button>
      </div>
    </div>
  );
}

export default List;