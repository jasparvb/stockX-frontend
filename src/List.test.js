import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from "react-redux";
import rootReducer from "./reducers/root";
import { BrowserRouter } from "react-router-dom";
import { createStore, applyMiddleware } from "redux";
import thunk from "redux-thunk";
import { composeWithDevTools } from 'redux-devtools-extension';
import List from './List';

const stocks = [
  {
    id: 1,
    ticker:"AAPL",
    name:"Apple Inc",
    list_id:1,
    quote: {
      open: 114.68,
      close: 112,
      change: -1.49,
      changePercent: -0.01313
    }
  }
];

const store = createStore(
  rootReducer,
  composeWithDevTools(
    applyMiddleware(thunk)
  )
);

it("renders without crashing", function() {
  render(
    <Provider store={store}>
      <BrowserRouter>
        <List stocks={stocks} />
      </BrowserRouter>
    </Provider>);
});

it("matches snapshot", function() {
  const { asFragment } = render(
    <Provider store={store}>
      <BrowserRouter>
        <List stocks={stocks} />
      </BrowserRouter>
    </Provider>
  );
  expect(asFragment()).toMatchSnapshot();
});
