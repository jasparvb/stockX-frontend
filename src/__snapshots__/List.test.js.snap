// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="List col-md-4 col-lg-4 mb-4"
  >
    <div
      class="list-box"
    >
      <h3
        class="mb-4"
      />
      <div
        class="ListItem px-5"
      >
        <a
          class="text-left col-md-5"
          href="/AAPL"
        >
          <h4
            class="mb-0"
          >
            AAPL
          </h4>
          <p>
            Apple Inc
          </p>
        </a>
        <div
          class="text-right col-md-5"
        >
          <h4
            class="mb-0"
          >
            $112.00
          </h4>
          <p
            class="percent-change down"
          >
            -1.31%
          </p>
        </div>
        <button
          class="btn remove"
        >
          <i
            class="fas fa-times"
          />
        </button>
      </div>
      <button
        class="btn btn-primary btn-sm mt-3"
      >
        Delete List
      </button>
    </div>
  </div>
</DocumentFragment>
`;
